ols/src/utils/token_handler.py

```diff
@@ -0,0 +1,94 @@
+"""Utility to handle tokens."""
+import tiktoken
+
+from ols.src.utils.constants import (
+    CONTEXT_WINDOW_LIMIT,
+    MINIMUM_CONTEXT_LIMIT,
+    RESPONSE_WINDOW_LIMIT,
+    TOKENIZER_MODEL,
+)
+from ols.utils.logger import Logger
+
+
+class TokenHandler:
+    """This class handles tokens.
+
+    Convert text to tokens.
+    Get rough estimation of token count.
+    Truncate text based on token limit.
+    """
+
+    def __init__(self):
+        """Initialize the class instance."""
+        self._logger = Logger("token_handler").logger
+        self._encoder = tiktoken.get_encoding(TOKENIZER_MODEL)
+
+    def text_to_tokens(self, text: str) -> list[int]:
+        """Convert text to tokens.
+
+        Args:
+            text: context text, ex: "This is my doc"
+
+        Returns:
+            List of tokens, ex: [1, 2, 3, 4]
+        """
+        # Note: We need an approximate tokens count.
+        # For different models, exact tokens count may vary.
+        return self._encoder.encode(text)
+
+    def tokens_to_text(self, tokens: list) -> str:
+        """Convert tokens to text.
+
+        Args:
+            tokens: ex: [1, 2, 3, 4]
+
+        Returns:
+            text: ex "This is my doc"
+        """
+        return self._encoder.decode(tokens)
+
+    def handle_context(self, retrieved_nodes, prompt_token_len: int) -> list[dict]:
```

Comment:
i think this will be more modular/easier to use if instead of taking the prompt token len as an arg, it just took a max length for the context.

The caller can decide how many tokens they have available or want to dedicate to RAG context content.  This function should just be responsible for returning content that is constrained to that desired length.